lib/matplotlib/backend_tools.py

```diff
@@ -0,0 +1,927 @@
+        """Set the cursor
+
+        This method has to be implemented per backend
+        """
+        pass
+
+
+class ToolCursorPosition(ToolBase):
+    """Send message with the current pointer position
+
+    This tool runs in the background reporting the position of the cursor
+    """
+    def __init__(self, *args, **kwargs):
+        ToolBase.__init__(self, *args, **kwargs)
+        self._idDrag = self.figure.canvas.mpl_connect(
+            'motion_notify_event', self.send_message)
+
+    def send_message(self, event):
+        """Call `matplotlib.backend_bases.NavigationBase.message_event`"""
+        if self.navigation.messagelock.locked():
+            return
+
+        message = ' '
+
+        if event.inaxes and event.inaxes.get_navigate():
+            try:
+                s = event.inaxes.format_coord(event.xdata, event.ydata)
+            except (ValueError, OverflowError):
+                pass
+            else:
+                message = s
+        self.navigation.message_event(message, self)
+
+
+class RubberbandBase(ToolBase):
+    """Draw and remove rubberband"""
+    def trigger(self, sender, event, data):
+        """Call `draw_rubberband` or `remove_rubberband` based on data"""
+        if not self.figure.canvas.widgetlock.available(sender):
+            return
+        if data is not None:
+            self.draw_rubberband(*data)
+        else:
+            self.remove_rubberband()
+
+    def draw_rubberband(self, *data):
+        """Draw rubberband
+
+        This method must get implemented per backend
+        """
+        pass
+
+    def remove_rubberband(self):
+        """Remove rubberband
+
+        This method must get implemented per backend
+        """
+        pass
+
+
+class ToolQuit(ToolBase):
+    """Tool to call the figure manager destroy method"""
+
+    description = 'Quit the figure'
+    keymap = rcParams['keymap.quit']
+
+    def trigger(self, sender, event, data=None):
+        Gcf.destroy_fig(self.figure)
+
+
+class ToolEnableAllNavigation(ToolBase):
+    """Tool to enable all axes for navigation interaction"""
+
+    description = 'Enables all axes navigation'
+    keymap = rcParams['keymap.all_axes']
+
+    def trigger(self, sender, event, data=None):
+        if event.inaxes is None:
+            return
+
+        for a in self.figure.get_axes():
+            if event.x is not None and event.y is not None \
+                    and a.in_axes(event):
+                a.set_navigate(True)
+
+
+class ToolEnableNavigation(ToolBase):
+    """Tool to enable a specific axes for navigation interaction"""
+
+    description = 'Enables one axes navigation'
+    keymap = (1, 2, 3, 4, 5, 6, 7, 8, 9)
+
+    def trigger(self, sender, event, data=None):
+        if event.inaxes is None:
+            return
+
+        n = int(event.key) - 1
+        for i, a in enumerate(self.figure.get_axes()):
+            # consider axes, in which the event was raised
+            # FIXME: Why only this axes?
```

Comment:
I would remove this comment. It is ancient and I think this feature is fairly well-established now.